### PR TITLE
layer: miscellaneous configuration changes

### PR DESCRIPTION
**Pull request by eric-ch**
Various small cleanups and refactoring in the layer:
- Use `KERNEL_MODULE_AUTOLOAD` where necessary instead of tweaking `/etc/modprobe.d/*.conf`
- Rename `recipe-core/eudev` -> `recipe-core/udev` to match upstream layers
- Remove `CONFIG_R8169` from dom0
- `99-kvm.rules` is very device specific. Distribution tend to handle that with a generic USB `usb-powersave.rules` ruleset for each device. Likely other devices could use that.
- dom0's persistent nic naming udev rule seems bogus and did not see to trigger anyway, remove it.

### CODE REVIEW

**Comment by eric-ch** [comment]
@jandryuk I have not seen anything turn out differently with the nics in the installer, but could use more tests. As I added to the commit message, I do not quite see how  this rule would help.

File: conf/machine/xenclient-dom0.conf [hunk 1]
```diff
@@ -16,3 +16,9 @@ module_conf_i915 = "options i915 enable_psr=0"
 KERNEL_MODULE_PROBECONF += " \
     i915 \
 "
+
+KERNEL_MODULE_AUTOLOAD += " \
+    psmouse \
+    xen-nefront \
```
[the remark this answers is not in the document]
Comment (jandryuk) [reply conf/machine/xenclient-dom0.conf:22]: I confirmed xen-netfront does autoload.

**Comment by jandryuk** [comment]
commit a570b92075a2c81708c780477adb641ce236395 used xenclient-dom0-tweaks to provide .inputrc because readline isn't installed unless debug-tweaks is active (or something else pulls in readline like python).  bash uses its own static copy of libreadline instead of the shared library.  So you have a file used by bash but provided by readline.  I'm not sure what to do about /etc/inputrc.

I think this PR should drop the inputrc change https://github.com/OpenXT/xenclient-oe/pull/1367/commits/14c0a93b9e45bf3362530e7cc15276d728153c7a and the rest can be merged.

**Comment by eric-ch** [comment]
@jandryuk That's unfortunate, but likely why it was shipped as a `${HOME}/.inputrc` and not the default `/etc/inputrc`. From what I can read/grep it seems this would belong to `base-file`. I'll see that in another PR so the rest can get in.

**Comment by eric-ch** [comment]
v3:
- Drop https://github.com/OpenXT/xenclient-oe/commit/14c0a93b9e45bf3362530e7cc15276d728153c7a

**Comment by eric-ch** [comment]
v2: [Build](https://openxt.ainfosec.com/buildbot/#/builders/14/builds/31): [$mcql-31](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom-quick-legacy/31/)

**Review by jandryuk — APPROVED** [review]

**Comment by eric-ch** [comment]
[Build](https://openxt.ainfosec.com/buildbot/#/builders/14/builds/32): [$mcql-32](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom-quick-legacy/32/)

**Comment by jandryuk** [comment]
Will merge soon

**Pull request merged**